File: swagger-client/csharp-dotnet2/src/main/CsharpDotNet2/mailbaby-client-csharp-dotnet2/Model/MailOrder.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// A mail service order record.  Each order represents one provisioned sending account with a dedicated SMTP username.  The &#x60;id&#x60; is the numeric identifier used across most API calls.  The &#x60;username&#x60; is always &#x60;mb&lt;id&gt;&#x60; and is the SMTP AUTH username for &#x60;relay.mailbaby.net&#x60;.
  /// </summary>
  [DataContract]
  public class MailOrder {
    /// <summary>
    /// The unique numeric ID of the mail order.  Used as the `id` parameter on sending calls, log queries, and stats queries.
    /// </summary>
    /// <value>The unique numeric ID of the mail order.  Used as the `id` parameter on sending calls, log queries, and stats queries.</value>
    [DataMember(Name="id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "id")]
    public int? Id { get; set; }

    /// <summary>
    /// The current order status.  Only `active` orders can be used for sending. `canceled` orders are retained for history but cannot send.
    /// </summary>
    /// <value>The current order status.  Only `active` orders can be used for sending. `canceled` orders are retained for history but cannot send.</value>
    [DataMember(Name="status", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "status")]
    public string Status { get; set; }

    /// <summary>
    /// The SMTP AUTH username for this order, always in the format `mb<id>`. Use together with the password from `GET /mail/{id}` to authenticate directly against `relay.mailbaby.net:25` if needed.
    /// </summary>
    /// <value>The SMTP AUTH username for this order, always in the format `mb<id>`. Use together with the password from `GET /mail/{id}` to authenticate directly against `relay.mailbaby.net:25` if needed.</value>
    [DataMember(Name="username", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "username")]
    public string Username { get; set; }

    /// <summary>
    /// Optional human-readable note associated with the order.
    /// </summary>
    /// <value>Optional human-readable note associated with the order.</value>
    [DataMember(Name="comment", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "comment")]
    public string Comment { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class MailOrder {\n");
      sb.Append("  Id: ").Append(Id).Append("\n");
      sb.Append("  Status: ").Append(Status).Append("\n");
      sb.Append("  Username: ").Append(Username).Append("\n");
      sb.Append("  Comment: ").Append(Comment).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: openapi-client/javascript-closure-angular/API/Client/MailOrder.js
goog.provide('API.Client.MailOrder');

/**
 * A mail service order record.  Each order represents one provisioned sending account with a dedicated SMTP username.  The `id` is the numeric identifier used across most API calls.  The `username` is always `mb<id>` and is the SMTP AUTH username for `relay.mailbaby.net`.
 * @record
 */
API.Client.MailOrder = function() {}

/**
 * The unique numeric ID of the mail order.  Used as the `id` parameter on sending calls, log queries, and stats queries.
 * @type {!number}
 * @export
 */
API.Client.MailOrder.prototype.id;

/**
 * The current order status.  Only `active` orders can be used for sending. `canceled` orders are retained for history but cannot send.
 * @type {!string}
 * @export
 */
API.Client.MailOrder.prototype.status;

/**
 * The SMTP AUTH username for this order, always in the format `mb<id>`. Use together with the password from `GET /mail/{id}` to authenticate directly against `relay.mailbaby.net:25` if needed.
 * @type {!string}
 * @export
 */
API.Client.MailOrder.prototype.username;

/**
 * Optional human-readable note associated with the order.
 * @type {!string}
 * @export
 */
API.Client.MailOrder.prototype.comment;

